### PR TITLE
Fix offset() for elements within foreign object context

### PR DESCRIPTION
**Pull request by kumilingus**
### Context

After inserting the table into SVG <foreignObject>, the cell selection boxes will stop appearing. The user can edit the cell, but the input element is hidden (or misplaced).

Selecting the `B3` cell:
image

Editing the `B3` cell:
image

I managed to locate the root cause of the problem. It lies within the DOM utility `offset` that does not account for the DOM element to be inside a foreign object. The foreign object has neither `offsetLeft` nor `offsetTop` properties defined and the `offset()` method ends up returning `{ top: NaN, left: NaN }`.

An example use case is the possibility to insert tables into an SVG document and connect columns or rows using links.
https://jsfiddle.net/kumilingus/j4s8bf97/

### How has this been tested?

1. A unit test has been added.
2. A patched version of _Handsontable_ with the same demo: https://jsfiddle.net/kumilingus/fq7zyr5b/

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

### Related issue(s):
1. https://github.com/handsontable/handsontable/issues/10480

### Affected project(s):
- [x] `handsontable`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)

### CODE REVIEW

**Comment by kumilingus** [comment]
It seems that your unit test environment does not allow the `offset` method to be tested? Perhaps that's the reason why there was no test for the method in the first place?

The DOM elements seem to have `offsetParent` always set to `null` and no size. Any suggestion how to test the issue?

**Comment by AMBudnik** [comment]
Hi @kumilingus 

I think that at this point, the PR is ready for a review. Thank you for sharing.

**Comment by budnix** [comment]
> The DOM elements seem to have `offsetParent` always set to `null` and no size. Any suggestion how to test the issue?

The `*.unit.js` files are run in a fake DOM environment (JSDOM), and the results may not be accurate. I'd suggest moving the test to the [element.spec.js](https://github.com/handsontable/handsontable/blob/develop/handsontable/src/helpers/dom/__tests__/element.spec.js) file. `*.spec.js` files are running in the real browser.

**Comment by kumilingus** [comment]
Hi @budnix, I moved the test to the `element.spec.js` file, but I wasn't sure how to run it. 🤦 😄

File: handsontable/src/helpers/dom/__tests__/element.spec.js [hunk 1]
```diff
@@ -22,4 +22,27 @@ describe('DOM helpers', () => {
...
+          <foreignObject width="50" height="50">
+            <div xmlns="http://www.w3.org/1999/xhtml">test</div>
+          </foreignObject>
+        </svg>
+      `;
+
+      document.body.appendChild(wrapper);
+
+      const element = wrapper.querySelector('div');
+      const elementOffset = offset(element);
```
Comment (budnix) [inline handsontable/src/helpers/dom/__tests__/element.spec.js:40]: I forgot to mention that in these tests, the DOM helpers are exposed in `Handsontable.dom` namespace. Should be:
```suggestion
      const elementOffset = Handsontable.dom.offset(element);
```

**Review by budnix — CHANGES_REQUESTED** [review]

**Comment by budnix** [comment]
> Hi @budnix, I moved the test to the `element.spec.js` file, but I wasn't sure how to run it. 🤦 😄

I reviewed the code besides one mistake it looks good 👌 

PS: You can run these tests by running `npm run test:e2e` (in the `handsontable` directory) or you can narrow them b `npm run test:e2e --testPathPattern=element`.

File: handsontable/src/helpers/dom/__tests__/element.spec.js (hunk 1 above)
Comment (kumilingus) [reply handsontable/src/helpers/dom/__tests__/element.spec.js]: Makes sense! Thanks.

**Comment by AMBudnik** [comment]
Hi @kumilingus 

I have great news! Your merged pull request was just published with Handsontable v14.0.0

Thank you again for your input.

Here's https://github.com/handsontable/handsontable/issues/9754 the full list of changes for this release.

**Pull request merged**